util/stmtsummary/statement_summary.go

```diff
@@ -349,21 +350,27 @@ func (ssMap *stmtSummaryByDigestMap) ToCurrentDatum(user *auth.UserIdentity, isS
 			rows = append(rows, record)
 		}
 	}
+	if otherDatum != nil {
+		rows = append(rows, otherDatum)
+	}
 	return rows
 }
 
 // ToHistoryDatum converts history statements summaries to datum.
 func (ssMap *stmtSummaryByDigestMap) ToHistoryDatum(user *auth.UserIdentity, isSuper bool) [][]types.Datum {
+	historySize := ssMap.historySize()
+
 	ssMap.Lock()
 	values := ssMap.summaryMap.Values()
+	otherDatum := ssMap.other.toHistoryDatum(historySize)
```

Comment:
It costs some time. You can copy `ssMap.other` in `ssMap.Lock`, but please call `toHistoryDatum` outside `ssMap.Lock`.

Reply:
ok